tools/spectral/ipa/rulesets/functions/eachPathAlternatesBetweenResourceNameAndPathParam.js

```diff
@@ -0,0 +1,33 @@
+import { isPathParam } from './utils/pathUtils';
+
+const ERROR_MESSAGE = 'API paths must alternate between resource name and path params.';
+const ERROR_RESULT = [{ message: ERROR_MESSAGE }];
+const AUTH_PREFIX = '/api/atlas/v2';
+const UNAUTH_PREFIX = '/api/atlas/v2/unauth';
+
+const getPrefix = (path) => {
+  if (path.includes(UNAUTH_PREFIX)) return UNAUTH_PREFIX;
+  if (path.includes(AUTH_PREFIX)) return AUTH_PREFIX;
+  return null;
+};
+
+const validatePathStructure = (elements) => {
+  return elements.every((element, index) => {
+    const isEvenIndex = index % 2 === 0;
+    return isEvenIndex ? !isPathParam(element) : isPathParam(element);
```

Comment:
```suggestion
    return isEvenIndex && !isPathParam(element);
```

Reply:
When the index is odd, it will return false this way. Am I missing anything?